osu.Game/Screens/OsuScreenStack.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Screens;
+using osu.Game.Graphics.Containers;
+
+namespace osu.Game.Screens
+{
+    public class OsuScreenStack : ScreenStack
+    {
+        [Cached]
+        private BackgroundScreenStack backgroundScreenStack;
+
+        private ParallaxContainer parallaxContainer;
+
+        protected float ParallaxAmount => parallaxContainer.ParallaxAmount;
+
+        public OsuScreenStack()
+        {
+            initializeStack();
+        }
+
+        public OsuScreenStack(IScreen baseScreen)
+            : base(baseScreen)
+        {
+            initializeStack();
+        }
+
+        private void initializeStack()
+        {
+            InternalChild = parallaxContainer = new ParallaxContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Child = backgroundScreenStack = new BackgroundScreenStack { RelativeSizeAxes = Axes.Both },
+            };
+
+            ScreenPushed += setParallax;
+            ScreenExited += setParallax;
+        }
+
+        private void setParallax(IScreen prev, IScreen next)
+        {
+            if (next != null)
```

Comment:
i think a better default would be `1` when next is null (since that what the initial value is when constructing the container).